src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1263,32 +1263,45 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
                     referencingArrayAddressMethod = targetMethod.IsArrayAddressMethod();
                 }
 
-                MethodDesc concreteMethod = targetMethod;
-                targetMethod = targetMethod.GetCanonMethodTarget(CanonicalFormKind.Specific);
-
                 pResult->kind = CORINFO_CALL_KIND.CORINFO_CALL;
 
-                if (targetMethod.IsConstructor && targetMethod.OwningType.IsString)
+                TypeDesc owningType = targetMethod.OwningType;
+                if (targetMethod.IsConstructor && owningType.IsString)
```

Comment:
IsConstructor is more expensive than IsString. It would be nice to reorder the checks.

Reply:
```suggestion
                if (owningType.IsString && targetMethod.IsConstructor)
```